From 127dc4251ea02365968996b6508c850943281422 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 14 Apr 2010 08:42:50 +0000
Subject: Do not ignore CD drives with blank media

Extend the special casing of CD drives in should_drive_be_ignored() to also
cover blank media. Otherwise we lose the drive for the blank media volume, and
with it the possibility to eject it.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=598554
Bug-Ubuntu: https://launchpad.net/bugs/425104
---
(limited to 'monitor/gdu/ggduvolumemonitor.c')

diff --git a/monitor/gdu/ggduvolumemonitor.c b/monitor/gdu/ggduvolumemonitor.c
index 4381d8b..446acfc 100644
--- a/monitor/gdu/ggduvolumemonitor.c
+++ b/monitor/gdu/ggduvolumemonitor.c
@@ -947,10 +947,11 @@ should_drive_be_ignored (GduPool *pool, GduDrive *d, GList *fstab_mount_points)
             ignored = TRUE;
         }
 
-      /* special case for audio discs: don't ignore the drive since we'll create
-       * a cdda:// mount for the drive
+      /* special case for audio and blank discs: don't ignore the drive since we'll create
+       * a cdda:// or burn:// mount for the drive
        */
-      if (gdu_device_is_optical_disc (device) && gdu_device_optical_disc_get_num_audio_tracks (device) > 0)
+      if (gdu_device_is_optical_disc (device) && (gdu_device_optical_disc_get_num_audio_tracks (device) > 0 ||
+                                                  gdu_device_optical_disc_get_is_blank (device)))
         {
           ignored = FALSE;
         }
--
cgit v0.9.0.2
